orttraining/orttraining/python/training/ortmodule/__init__.py

```diff
@@ -4,57 +4,41 @@
-
-if torch_extensions_dir is not None and os.access(torch_extensions_dir, os.X_OK | os.W_OK):
-    TORCH_CPP_BUILD_DIR = torch_extensions_dir
-elif not os.access(python_package_dir, os.X_OK | os.W_OK):
-    if os.access(home_dir, os.X_OK | os.W_OK):
-        TORCH_CPP_BUILD_DIR = TORCH_CPP_BUILD_DIR_BACKUP
-    else:
-        extra_message = ''
-        if torch_extensions_dir:
-            extra_message = 'or the path pointed by the TORCH_EXTENSIONS_DIR environment variable '
-        raise PermissionError('ORTModule could not find a writable directory to cache its internal files.',
-                              f'Make {python_package_dir} or {home_dir} {extra_message}writable and try again.')
-
-# Check whether Torch C++ extension compilation was aborted in previous runs
-if not os.path.exists(TORCH_CPP_BUILD_DIR):
-    os.makedirs(TORCH_CPP_BUILD_DIR, exist_ok = True)
-elif os.path.exists(os.path.join(TORCH_CPP_BUILD_DIR,'lock')):
-    print("WARNING: ORTModule detected PyTorch's CPP extension lock file during initialization, "
-          "which can cause the script to stop responding. "
-          f"Delete {os.path.join(TORCH_CPP_BUILD_DIR,'lock')} if a hang occurs.")
-
-# Verify proper PyTorch is installed before proceding to ONNX Runtime initialization
+# Verify minimum PyTorch version is installed before proceding to ONNX Runtime initialization
 try:
     import torch
-    torch_version = version.parse(torch.__version__.split('+')[0])
-    minimum_torch_version = version.parse(MINIMUM_TORCH_VERSION_STR)
-    if torch_version < minimum_torch_version:
+    runtime_pytorch_version = version.parse(torch.__version__.split('+')[0])
+    minimum_runtime_pytorch_version = version.parse(MINIMUM_RUNTIME_PYTORCH_VERSION_STR)
+    if runtime_pytorch_version < minimum_runtime_pytorch_version:
         raise RuntimeError(
-            f'ONNX Runtime ORTModule frontend requires PyTorch version greater or equal to {MINIMUM_TORCH_VERSION_STR}, '
+            f'ONNX Runtime ORTModule frontend requires PyTorch version greater or equal to {MINIMUM_RUNTIME_PYTORCH_VERSION_STR}, '
             f'but version {torch.__version__} was found instead.')
 except:
-    raise(f'PyTorch {MINIMUM_TORCH_VERSION_STR} must be installed in order to run ONNX Runtime ORTModule frontend!')
+    raise RuntimeError(f'PyTorch {MINIMUM_RUNTIME_PYTORCH_VERSION_STR} must be installed in order to run ONNX Runtime ORTModule frontend!')
+
+# Verify whether PyTorch C++ extensions are already compiled
+torch_cpp_exts = glob(os.path.join(TORCH_CPP_DIR, '*.so'))
+torch_cpp_exts.extend(glob(os.path.join(TORCH_CPP_DIR, '*.dll')))
+torch_cpp_exts.extend(glob(os.path.join(TORCH_CPP_DIR, '*.dylib')))
+if not torch_cpp_exts and '-m' not in sys.argv:
+    raise EnvironmentError(f"ORTModule's extensions were not detected at '{TORCH_CPP_DIR}' folder. "
+                           "Run `python -m torch_ort.configure` before using `ORTModule` frontend.")
```

Comment:
>python -m torch_ort.configure [](http://example.com/codeflow?start=33&length=29)

seems misleading, should it be "python -m onnxruntime.training.ortmodule.torch_cpp_extensions.install"?  I had to run the latter for zcode to work - FYI

Reply:
Pubicly ORTModule is only used through torch_ort, so we give instructions using torch_ort namespace. Internally it is possible to bypass torch_ort by using onnxruntime, but we dont want to make this the public contract

Reply:
I'll keep an eye on torch_ort release and verify it again.